torchgeo/datasets/geo.py

```diff
@@ -437,25 +437,17 @@ def __init__(
         filepaths = []
         datetimes = []
         geometries = []
+        vrt = None
```

Comment:
I'm wondering if this is the right place to put this, or if it should be further in. If something goes wrong in the loop, we will break out and try to close `vrt`. But if something goes wrong twice, `vrt` will already be closed.